test/extended/testdata/gssapi/fedora/base/gssapi-tests.sh

```diff
@@ -0,0 +1,280 @@
+#!/usr/bin/bash
+
+set -e
+set -x
+
+echo "I ran this on `uname -a`"
+
+os::test::junit::declare_suite_start "test-extended/gssapiproxy-tests"
+
+CLIENT_HAS_GSSAPI=true
+SERVER_HAS_BASIC=true
+users=(user1 user2 user3 user4 user5)
+
+# Client has no GSSAPI and server is GSSAPI only
+# Everything fails
+
+if [[ -z CLIENT_HAS_GSSAPI && -z SERVER_HAS_BASIC ]]; then
+    for u in "${users[@]}"; do
+        full="$u$realm"
+        os::cmd::expect_failure_and_text 'oc login' 'No Kerberos credentials available'
+        os::cmd::expect_failure_and_not_text 'oc whoami' $u
```

Comment:
I think we actually have three client cases:
1. no gssapi libs
2. gssapi libs, unconfigured to KDC
3. gssapi libs, configured to KDC

no gssapi libs, no basic on server:
```
os::cmd::expect_failure_and_text "oc login" "Unauthorized"
os::cmd::expect_failure_and_text "oc login -u user1" "Unauthorized"
os::cmd::expect_failure_and_text "oc login -u user1 -p password" "Unauthorized"
os::cmd::expect_failure_and_text "oc login -u user1@$realm" "Unauthorized"
os::cmd::expect_failure_and_text "oc login -u user1@$realm -p password" "Unauthorized"
```

gssapi libs present but unconfigured, no basic on server:
```
os::cmd::expect_failure_and_text "oc login" "No Kerberos credentials available"
os::cmd::expect_failure_and_text "oc login -u user1" "No Kerberos credentials available"
os::cmd::expect_failure_and_text "oc login -u user1 -p password" "No Kerberos credentials available"
os::cmd::expect_failure_and_text "oc login -u user1@$realm" "No Kerberos credentials available"
os::cmd::expect_failure_and_text "oc login -u user1@$realm -p password" "No Kerberos credentials available"
```